cmds/tui.c

```diff
@@ -695,6 +718,7 @@ static void build_partial_graph(struct tui_report_node *root_node,
 	}
 
 	tui_window_init(&graph->win, &graph_ops);
+	graph->win.last_index = tui_last_index(&graph->win);
```

Comment:
At `tui_graph_init`, 
it calls `tui_window_init` for `normal graph` and `partial graph`.

since at this point, there isn't any selected node for `partial graph`, 
so `tui_last_index()` will cause an error.

```
static void tui_window_init(struct tui_window *win,
                            const struct tui_window_ops *ops)
{
        void *top = ops->top(win, true);

        win->ops = ops;
        win->top = top;
        win->curr = win->old = top;
        win->top_index = win->curr_index = 0;
        win->last_index = tui_last_index(&graph->win);
}
```

Reply:
I don't see what the problem is.  In the last statement you should pass `win` not the `&graph->win`.

Reply:
It's my bad with upper code snippet. (Mistyped `win` as `&graph-win`)

[tui_graph_last_index_move_to_window_init.patch.txt](https://github.com/namhyung/uftrace/files/3029632/tui_graph_last_index_move_to_window_init.patch.txt)

If you apply this patch, you can find out it will cause an SIGSEGV error. (which is segment fault)

```
Building graph for TUI...fish: “uftrace tui” terminated by signal SIGSEGV (Address boundary error)
```

This is caused by `partial graph`. 

https://github.com/namhyung/uftrace/blob/master/cmds/tui.c#L567-L606

At line 597, `partial graph` will call tui_window_init.
With the upper patch applied, `tui_last_index` will be called for `partial graph` without any node selected.

And this will eventually cause an segment fault.

Reply:
Could you please make it not crashed?  Maybe you need to call `INIT_LIST_HEAD()` before `tui_window_init()`.

Reply:
Seems working!